subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -392,13 +392,27 @@ class AbstractIntegrationSpec extends Specification {
         result.assertHasPostBuildOutput(string.trim())
     }
 
-    void useRepositoryMirrors() {
+    void useRepositoryMirrors(boolean global = false) {
+        if (System.getenv("CI") == null) {
+            return
+        }
+
         executer.beforeExecute {
-            executer.usingInitScript(RepoScriptBlockUtil.createMirrorInitScript())
+            if (global) {
+                File userHome = file('user-home')
+                executer.withGradleUserHomeDir(userHome)
+                file('user-home/init.gradle').text = RepoScriptBlockUtil.mirrorInitScript()
```

Comment:
I think we should add the script into `user-home/init.d/mirrors.gradle`. So we don't break tests which write `init.gradle`.

Reply:
Done.